### ISSUE TITLE
Recursion support

### ISSUE THREAD

**Issue by teran-mckinney** [logrotate/logrotate#5]
While in logrotate.conf you can do /var/log/*.log /var/log/*/*.log /var/log/*/*/*.log {} to setup a default catch-all for logs, I'd rather see a recursion option that you could put under /var/log/*.log that would recursively look for *.log anywhere in /var/log/, no matter how many directories deep (within reason). It seems a bit cleaner to me that way.

Example:

```
/var/log/*.log {
    copytruncate
    daily
    compress
    recurse
}
```

**Comment by rvealokin** [comment]
`logrotate` uses `glob` internally so an easy workaround could be enabling `globstar` if using Bash 4 or higher. This should allow to use the following syntax `/var/log/**/*.log`.

The `globstar` can be enabled with the following command:

```
shopt -s globstar
```

Some details on `globstar` option: http://www.linuxjournal.com/content/globstar-new-bash-globbing-option

**Comment by teran-mckinney** [comment]
Ah, thank you! I don't think these boxes are on bash 4.X, but that's certainly an option down the road. Didn't realize logrotate used bash in such a way.

So in this case, I would begin the logrotate configuration with `shopt -s globstar`. Is that correct?

**Comment by rvealokin** [comment]
No, you should not put this into config. It's the command to enable `globstar` option in your shell. I haven't tried it myself with `logrotate` yet though.

**Comment by LorenzoAncora** [comment]
@inikolaev I have the same issue and enabling `globstar` does not work: logrotate fails with `verifying log file path failed`, because the matching ends at `**`.

(logrotate 3.11.0, bash 4.4.23)

**Comment by tmccombs** [comment]
@inikolaev that workaround doesn't work. afaik, logrotate uses the glob(3) c funtion to do expansion, not bash. so setting an option in bash has no effect on logrotate.

**Comment by inikolaev** [comment]
> @inikolaev that workaround doesn't work. afaik, logrotate uses the glob(3) c funtion to do expansion, not bash. so setting an option in bash has no effect on logrotate.

You might be right - I haven't realized that those are two different globs.

**Comment by kdudka** [comment]
Shell options for globbing only affect how the shell interpreter itself expands glob while interpreting shell code.  It cannot affect globbing done by external binary programs in any way.

**Comment by Crapshit** [comment]
Same here on RHEL 7 with logrotate 3.8.6 and bash 4.2.46(2)-release.
@kdudka: It would be great to have a [feature](https://github.com/logrotate/logrotate/issues/177#issuecomment-405121807) as mentioned from @LorenzoAncora in issue #177
